src/core/handle.ml

```diff
@@ -245,7 +259,61 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
       (* Actually add the symbol to the signature. *)
       out 3 "(symb) %s ≔ %a\n" x.elt pp_term t;
       let d = if op then None else Some(t) in
-      let ss = Sig_state.add_symbol ss expo Defin Eager x a impl d in
+      (fst (Sig_state.add_symbol ss expo Defin Eager x a impl d), None)
+  | P_inductive(ms, id, a, l)     ->
+      (* Verify the modifiers. *)
+      let (prop, e, mstrat) = handle_modifiers ms in
+      if prop <> Defin then
+        fatal cmd.pos "Property modifiers cannot be used in inductive types.";
+      if mstrat <> Eager then
+        fatal cmd.pos "Pattern matching strategy modifiers cannot be used \
+                       in inductive types.";
+      (* Add the inductive type in the signature *)
+      let (ss, sym_typ) = handle_symbol ss e Injec Eager id [] a in
+      (* Add the constructors in the signature.  *)
+      let add_cons :
+            sig_state * sym list -> ident * p_term -> sig_state * sym list
+        = fun(ss, cons_list) (id, a) ->
+        let (ss, sym_cons) = handle_symbol ss e Const Eager id [] a in
+        (ss, sym_cons::cons_list)
+      in
+      let (ss, cons_list) = List.fold_left add_cons (ss, []) l in
+      let cons_list = List.rev cons_list  in
+      (* Compute the induction principle *)
+      let ind_typ = Inductive.principle ss cmd.pos sym_typ cons_list in
+      (* Check the type of the induction principle *)
+      (match Typing.infer [] ind_typ with
+       | Some _ -> ()
+       | None   ->
+           fatal cmd.pos "The type of the generated inductive principle of
+                          [%a] isn't correct. Please, raise an issue."
+             pp_term ind_typ);
+      (* Add the induction principle in the signature *)
+      let ind_name = Pos.make cmd.pos (Parser.add_prefix "ind_" id.elt) in
+      if StrSet.mem id.elt !(ss.signature.sign_idents) then
+        begin
+        Sign.add_ident ss.signature ind_name.elt;
+        if !log_enabled then
+          log_hndl "This ident %a is already in the signature \
+                    and now the ident %a too !"
+            Pretty.pp_ident id Pretty.pp_ident ind_name;
+        end;
+      let (ss, sym_ind) =
+        Sig_state.add_symbol ss e Defin Eager ind_name ind_typ [] None
+      in
+      if !log_enabled then
+        log_hndl "The induction principle named %a is %a"
+          Pretty.pp_ident ind_name Print.pp_term ind_typ;
+      (* Compute the rules associated with the induction principle *)
+      let rs =
+        Inductive.ind_rule sym_typ sym_ind cons_list
+      in
+      let (ss, ind_rules) = handle_rules ss rs       in
+      let ind_rules = List.map (fun (_,r) -> r.elt) ind_rules in
+      (* Add the rules of induction principle in the signature *)
+      sym_ind.sym_rules := ind_rules @ !(sym_ind.sym_rules);
```

Comment:
This line is useless and wrong as rules are already added by handle_rules.

Reply:
So the final result is:
`let (ss, _) = handle_rules ss rs in`
that's mean I should come back and delete the function `handle_rules`. I mean, the second argument is never used.

Reply:
So the final result is:
`let (ss, _) = handle_rules ss rs in`
that's mean I should come back and delete the function `handle_rules`. I mean, the second argument is never used.

Reply:
indeed, you can change the output of handle_rules by removing the list of rules if it is used nowhere

Reply:
OK, done.